subsys/spm/secure_services.c

```diff
@@ -52,6 +52,23 @@ int spm_secure_services_init(void)
 	return err;
 }
 
+#ifdef CONFIG_SPM_SERVICE_READ_FICR
+__TZ_NONSECURE_ENTRY_FUNC
+int spm_request_read_ficr(u16_t offset, u32_t *destination)
+{
+	if (destination == NULL) {
+		return -EINVAL;
+	}
+
+	if ((offset >= 0x204 && offset < 0xC20) ||
+		(offset >= 0x130 && offset <= 0x134)) {
```

Comment:
This has the same purpose as the functions which checks FICR in system_nrf9160.c (ref: [here](https://github.com/NordicPlayground/fw-nrfconnect-zephyr/blob/1e34490406b4231fa79558d2759264397fbefa4c/ext/hal/nordic/nrfx/mdk/system_nrf9160.c#L183)). Reading the 4-byte register with offset 0x134 is intentional. If it is OK in system_nrf9160.c, then it should also be OK here.

u32_t * already implies that this should be 4-byte aligned. If the user is providing something odd, the final behavior will be governed by Coretex-M33.

You reminded me of one more thing: 0xC20 should be changed to 0xC1C. This is to make sure access to anything beyond 0xC20 will be discarded. This will be scheduled to the next rework.